python_language/filter_words/filter_words.py

```diff
@@ -0,0 +1,38 @@
+"""
+Filter Words
+------------
+
+Print out only words that start with "o", ignoring case::
+
+    lyrics = '''My Bonnie lies over the ocean.
+                My Bonnie lies over the sea.
+                My Bonnie lies over the ocean.
+                Oh bring back my Bonnie to me.
+                '''
+
+Bonus points: print out words only once.
+
+See :ref:`filter-words-solution`.
+"""
+import re
+lyrics = '''My Bonnie lies over the ocean.
+            My Bonnie lies over the sea.
+            My Bonnie lies over the ocean.
+            Oh bring back my Bonnie to me.
+         '''
+lyrics=lyrics.lower()
+lyrics=re.sub(r'[\.,]?','',lyrics)
+lyricList=lyrics.split()
+owords=[]
+for word in lyricList:
+        if(word[:1]=='o'):
+            try:
+                x=owords.index(word)
+            except:
```

Comment:
I'm always a little wary of `except` that has no specific `Exception` that it is looking for.  I _think_ in this case you are checking to see whether `word` is already in `owords`.  The error that you would want to catch here would be a `ValueError`.  When you specify `except ValueError`, then you catch that but still let other kinds of errors stop execution if they should.

Better yet, you can just do a simple `if word not in owords:` clause, which is a little more readable.